lib/libesp32/berry_tasmota/src/be_ULP_lib.c

```diff
@@ -0,0 +1,55 @@
+/********************************************************************
+ * Tasmota lib
+ * 
+ * To use: import ULP`
+ *******************************************************************/
+#include "be_constobj.h"
+#include "be_mapping.h"
+
+#if defined(USE_BERRY_ULP) && defined(CONFIG_IDF_TARGET_ESP32)
+
+#include "esp32/ulp.h"
+#include "driver/rtc_io.h"
+#include "driver/gpio.h"
+#include "driver/adc.h"
+
+extern void be_ULP_run(int32_t entry);
+BE_FUNC_CTYPE_DECLARE(be_ULP_run, "", "[i]");
+
+extern void be_ULP_wake_up_period(int32_t period_index, int32_t period_us);
+BE_FUNC_CTYPE_DECLARE(be_ULP_wake_up_period, "", "ii");
+
+extern int32_t be_ULP_set_mem(int32_t pos, int32_t value);
+BE_FUNC_CTYPE_DECLARE(be_ULP_set_mem, "i", "ii");
+
+extern int32_t be_ULP_get_mem(int32_t pos);
+BE_FUNC_CTYPE_DECLARE(be_ULP_get_mem, "i", "i");
+
+extern int32_t be_ULP_gpio_init(gpio_num_t pin, rtc_gpio_mode_t mode);
+BE_FUNC_CTYPE_DECLARE(be_ULP_gpio_init, "i", "ii");
+
+extern void be_ULP_adc_config(struct bvm *vm, adc1_channel_t channel, adc_atten_t attenuation, adc_bits_width_t width);
+BE_FUNC_CTYPE_DECLARE(be_ULP_adc_config, "", "iii");
```

Comment:
Sorry one more change, should be:

`BE_FUNC_CTYPE_DECLARE(be_ULP_adc_config, "", "@iii");`

`@` indicates to add `vm` argument

Reply:
This is the most educational PR for me so far 😀 .